fury/gltf.py

```diff
@@ -0,0 +1,398 @@
+import numpy as np
+import base64
+import json as j
+import os
+from dataclasses import dataclass, asdict
+from fury.lib import PNGReader, Texture, JPEGReader, ImageFlip, PolyData
+from fury import window, transform, utils
+from io import BytesIO
+
+
+comp_type = {
+    5120: {'size': 1, 'dtype': np.byte},
+    5121: {'size': 1, 'dtype': np.ubyte},
+    5122: {'size': 2, 'dtype': np.short},
+    5123: {'size': 2, 'dtype': np.ushort},
+    5125: {'size': 4, 'dtype': np.uint},
+    5126: {'size': 4, 'dtype': np.float32}
+}
+
+acc_type = {
+    'SCALAR': 3,
+    'VEC2': 2,
+    'VEC3': 3,
+    'VEC4': 4
+}
+
+
+@dataclass
+class Node:
+    name: str = None
+    camera: int = None
+    skin: int = None
+    mesh: int = None
+    matrix: np.ndarray = None
+    rotation: np.ndarray = None
+    translation: np.ndarray = None
+    scale: np.ndarray = None
+    children: np.ndarray = None
+
+
+@dataclass
+class Mesh:
+    primitives: np.ndarray
+    name: str = None
+
+
+@dataclass
+class Material:
+    pbrMetallicRoughness: dict = None
+    occlusionTexture: dict = None
+    normalTexture: dict = None
+    emissiveTexture: dict = None
+    emissiveFactor: list[int] = None
+    alphaMode: str = None
+    alphaCutoff: float = None
+    doubleSided: bool = False
+    name: str = None
+    # vtkTextures and Materials
+    baseColorTexture: Texture = None
+    metallicRoughnessTexture: Texture = None
+
+
+@dataclass
+class Primitive:
+    attributes: dict = None
+    indices: int = None
+    mode: int = 4
+    target: any = None
+    polydata: PolyData = None
+    material: Material = None
+
+    @classmethod
+    def set_polydata(cls, polydata):
+        cls.polydata = polydata
+        return Primitive(polydata=polydata)
+
+
+@dataclass
+class Perspective:
+    aspectRatio: float = None
+    yfov: float = None
+    zfar: float = None
+    znear: float = None
+
+
+@dataclass
+class Orthographic:
+    xmag: float = None
+    ymag: float = None
+    zfar: float = None
+    znear: float = None
+
+
+@dataclass
+class Camera:
+    type: str
+    perspective: Perspective = None
+    orthographic: Orthographic = None
+
+
+@dataclass
+class Accessor:
+    componentType: int
+    type: str
+    bufferView: int = None
+    byteOffset: int = 0
+    normalized: bool = False
+    count: int = False
+    max: list = False
+    min: list = False
+    sparse: any = None
+    name: str = None
+
+
+@dataclass
+class BufferView:
+    buffer: int
+    byteLength: int
+    byteOffset: int = 0
+    byteStride: int = None
+    target: int = None
+    name: str = None
+
+
+@dataclass
+class Buffer:
+    byteLength: int
+    uri: str = None
+    name: str = None
+
+
+@dataclass
+class Scene:
+    nodes: list[int] = [0]
```

Comment:
use `default_factory` as suggested by the error `ValueError: mutable default <class 'list'> for field nodes is not allowed: use default_factory
`

Reply:
Thanks for the wonderful review @m-agour, I didn't know it was a python>=3.9 feature. I will look into it.

Reply:
+1 Getting the same error on Python 3.9.13

Reply:
Hi @m-agour and @ganimtron-10 , inorder to type hint lists you need to do the following:
```python
from typing import List
nodes: List[int] = [0]
```

Reply:
Works on all existing python versions